### DISCUSSION TITLE
[BUG] 添加直播间出错 aiohttp.client_exceptions.ClientResponseError: 412

### DISCUSSION THREAD

**Discussion by ETChush** [timerring/bilive#312] [Q&A] [answered]
黑群晖docker中部署的Ubuntu 22.04.4
src/config.py中修改
GPU_EXIST 参数设置为 False
MODEL_TYPE 调整为 merge
如何根据步骤走
第五步. 启动自动录制后打开录制面板
添加直播间出错
![Snipaste_2024-12-27_11-50-59](https://github.com/user-attachments/assets/d12da96d-6571-4f86-9835-35ce5fcfbfdf)

aiohttp.client_exceptions.ClientResponseError: 412, message='Precondition Failed', url='https://api.live.bilibili.com/room/v1/Room/room_init?id=9425503'

错误类型: aiohttp.client_exceptions.ClientResponseError

错误代码: 412 Precondition Failed

错误 URL: https://api.live.bilibili.com/room/v1/Room/room_init?id=9425503

错误描述: 服务器返回了 412 错误，表示请求未满足某些前置条件。

**Comment by timerring** [comment, marked as answer]
录制部分我采用的是 [`blrec`](https://github.com/acgnhiki/blrec)，根据 [相应的issue](https://github.com/acgnhiki/blrec/pull/264) 情况疑似被风控了。这是[请求的B站 api 报错 412 情况](https://github.com/SocialSisterYi/bilibili-API-collect/issues/872)。

因此我建议添加或者更换 `UA`，还可以考虑 添加 cookies，等待部分时间执行`./record.sh`重试。

另外，我在本地尝试了一下（默认 UA 无 cookies）， api 添加该房间是正常的。
![image](https://github.com/user-attachments/assets/cdcce662-b89a-42c0-ba78-435004b372fa)

**Comment by ETChush** [comment]
还是不行，后面我用docker单独部署了blrec，可以正常添加直播间并录制，我看下更改保存目录到bilive路径下能不能实现曲线救国，能否进行自动压制并上传

**Comment by timerring** [comment]
这样肯定没问题，因为录制和处理和上传三个进程是分开的，保证视频录制的路径在 bilive 指定的路径下就可以。

不过还是建议一段时间后重新 kill 了进程重启`./record.sh`试一下 pip 的 blrec。因为本质上这两种方式都是同一个代码，用的也都是用的同一个 b 站 api，应该不会存在一个能用而另一个不能用的情况。
